src/abstract_block.jl

```diff
@@ -363,3 +370,37 @@ function parameters_range!(out::Vector{Tuple{T,T}}, block::AbstractBlock) where
         parameters_range!(out, subblock)
     end
 end
+
+# non-inplace versions
+"""
+    apply(register, block)
+
+The non-inplace version of applying a block (of quantum circuit) to a quantum register.
+Check `apply!` for the faster inplace version.
+"""
+apply(r::AbstractRegister, b) = apply!(copy(r), b)
```

Comment:
should we have this in YaoAPI or not? I'm still thinking, prob not, since I don't think this is an API gonna be used elsewhere

Reply:
Let us keep the set of interfaces in YaoAPI small. Let's only do it when nessesary.